src/library/objects/ShowcaseExporter.js

```diff
@@ -0,0 +1,873 @@
+        if (nameLines.length === 1) {
+            if (!fake)
+                ctx.fillText(nameLines[0], x, y + fontSize);
+            y += fontSize + 5;
+        } else {
+            for (var i = 0; i < nameLines.length; i++) {
+                if (!fake)
+                    ctx.fillText(nameLines[i], x, y + fontSize);
+                y += fontSize + 5;
+            }
+        }
+        return y;
+    };
+
+    ShowcaseExporter.prototype._splitText = function (text, ctx, maxWidth, splitter) {
+        if (typeof splitter === "undefined")
+            splitter = " ";
+
+        if (text.indexOf(splitter) === -1)
+            splitter = "";
+
+        var rows = [];
+        var words = text.split(splitter);
+
+        //@TODO what if single word will be so long so it goes outside maxWidth?
+        while (words.length > 0) {
+            var line = [];
+            var next = words.shift();
+            while (ctx.measureText((line.join(splitter) + splitter + next).trim()).width < maxWidth && next !== "") {
+                line.push(next);
+                if (words.length > 0)
+                    next = words.shift();
+                else
+                    next = "";
+            }
+            rows.push(line.join(splitter));
+            if (next !== "") {
+                words.unshift(next);
+            }
+        }
+
+        return rows;
+    };
+
+    ShowcaseExporter.prototype._drawEquipStats = function (equip, ctx, x, y, available, fake) {
+        var fontSize = 10;
+        ctx.font = fontSize + "pt \"Helvetica Neue\", Helvetica, Arial, sans-serif";
+        ctx.fillStyle = "#000";
+
+        var startX = x;
+        var img;
+        var noStats = true;
+
+        function drawStat(img) {
+            if (typeof equip[i] !== "undefined" && equip[i] > 0) {
+                noStats = false;
+                if (x + img.width + 3 + ctx.measureText(equip[i]).width > startX + available) {
+                    y += img.height + 5;
+                    x = startX;
+                }
+
+                if (!fake)
+                    ctx.drawImage(img, 0, 0, img.width, img.height, x, y, img.width, img.height);
+                x += img.width + 3;
+                if (!fake)
+                    ctx.fillText(equip[i], x, y + (img.height + fontSize) / 2);
+                x += ctx.measureText(equip[i]).width;
+                x += 10;
+            }
+        }
+
+        var sortingParam = KC3StrategyTabs.gears.definition._defaultCompareMethod["t" + KC3Master.slotitem(equip.masterId).api_type[3]];
+        var i;
+        if (!!sortingParam && sortingParam !== "overall") {
+            i = sortingParam;
+            img = this._statsImages[i];
+            drawStat.call(this, img);
+        }
+
+
+        for (i in this._statsImages) {
+            if (!this._statsImages.hasOwnProperty(i))
+                continue;
+
+            if (i === "or" && this.aircraftTypes.indexOf(KC3Master.slotitem(equip.masterId).api_type[3]) === -1)
```

Comment:
seems not work as `this.aircraftTypes` will become `[]`. just use `KC3GearManager.landBasedAircraftType3Ids` will do. `api_distance` is not suit for all carrier based aircraft: type 21/22 are not able to deploy to land base.